oak_restricted_kernel/src/syscall/mmap.rs

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::{
+    mm::{Mapper, PageTableFlags},
+    FRAME_ALLOCATOR, PAGE_TABLES,
+};
+use core::{
+    cmp::max,
+    ffi::{c_int, c_size_t, c_void},
+    iter::repeat,
+    ops::DerefMut,
+    ptr::NonNull,
+};
+use oak_restricted_kernel_interface::{
+    syscalls::{MmapFlags, MmapProtection},
+    Errno,
+};
+use x86_64::{
+    align_up,
+    structures::paging::{FrameAllocator, Page, PageSize, Size2MiB},
+    VirtAddr,
+};
+
+fn mmap(
+    addr: *const c_void,
+    size: usize,
+    prot: MmapProtection,
+    flags: MmapFlags,
+) -> Result<NonNull<c_void>, Errno> {
+    if size == 0 {
+        log::warn!("invalid size passed to mmap: {}", size);
+        return Err(Errno::EINVAL);
+    }
+
+    if !flags.contains(MmapFlags::MAP_ANONYMOUS) || !flags.contains(MmapFlags::MAP_PRIVATE) {
+        log::warn!("invalid flags passed to mmap: {:?}", flags);
+        return Err(Errno::EINVAL);
+    }
+
+    // even if addr is below 2 MiB, don't touch anything below that (we don't want to make 0x0 a
```

Comment:
super nit: comments should start with a capital and end with a full stop.

Reply:
Done.